server/src/main/kotlin/fi/vauhtijuoksu/vauhtijuoksuapi/server/impl/StreamMetadataRouter.kt

```diff
@@ -34,69 +43,153 @@ class StreamMetadataRouter
             "donation_goal": null,
             "current_game_id": null,
             "donatebar_info": [],
-            "counters": []
+            "counters": [],
+            "heart_rates": [],
+            "timers": []
         }"""
     )
+    private val emptyMetaData = StreamMetadata(
+        null,
+        null,
+        listOf(),
+        listOf(),
+        listOf()
+    )
+    private val logger = KotlinLogging.logger {}
 
     @Suppress("SwallowedException") // Not swallowed, the message is used in augmented error
     fun configure(router: Router) {
         router.route("/stream-metadata").handler { ctx ->
             ctx.response().putHeader("content-type", "application/json")
             ctx.next()
         }
+
+        handleGet(router)
+        handlePatch(router)
+    }
+
+    private fun handleGet(router: Router) {
         router.get("/stream-metadata")
             .handler(publicEndpointCorsHandler)
             .handler { ctx ->
-                db.get()
-                    .onSuccess {
+                var metaData = emptyMetaData
+                dbStream.get()
+                    .onSuccess { metaData = it }
```

Comment:
I'm not a fan of this approach. It feels a bit like global variable, and breaks the flow of futures. In a similar situation with incentivecodes, I'm trying a data class with required intermediate values. 

If you feel like trying it, refactoring this to use something like this might be worthwhile. 
```
data class StreamMetaDataWithTimers(
  val metadata: StreamMetadata,
  val timers: List<Timer>,
)
```

But I don't have a working solution to point to AtM, and it's a small code block so it's fine like this is you don't feel like experimenting

Reply:
Issue here was that I could not find a way to pass metadata throug to the compose. Having a new global StreamMetaDataWithTimers class does not seem much better solution

Reply:
The class doesn't have to be global. Let's keep it like this and I'll show you my idea with incentive codes once I get that done